zone/mob.cpp

```diff
@@ -4887,6 +4887,36 @@ void Mob::RemoveNimbusEffect(int effectid)
 	safe_delete(outapp);
 }
 
+void Mob::RemoveAllNimbusEffects()
+{
+	auto current_nimbus_one = nimbus_effect1;
```

Comment:
This can be simplified

```cpp
	std::vector<uint32> nimbi = {nimbus_effect1, nimbus_effect2, nimbus_effect3};

	for (auto &n: nimbi) {
		auto p  = new EQApplicationPacket(
			OP_RemoveNimbusEffect,
			sizeof(RemoveNimbusEffect_Struct)
		);
		auto *r = (RemoveNimbusEffect_Struct *) p->pBuffer;
		r->spawnid       = GetID();
		r->nimbus_effect = (int32) n;
		entity_list.QueueClients(this, p);
		safe_delete(p);
	}
```

Reply:
Probably better to use a normal C style array here instead, don't have to worry about the compiler optimizing out the dynamic memory allocations vectors probably introduce.

Reply:
You can just set them all to 0 after the loop or you can create a pointer to the mob attributes within the vector so you can modify it within the loop

**Example**

```cpp
	std::vector<uint32*> nimbi = {&nimbus_effect1, &nimbus_effect2, &nimbus_effect3};

	for (auto &n: nimbi) {
		auto p  = new EQApplicationPacket(
			OP_RemoveNimbusEffect,
			sizeof(RemoveNimbusEffect_Struct)
		);
		auto *r = (RemoveNimbusEffect_Struct *) p->pBuffer;
		r->spawnid       = GetID();
		r->nimbus_effect = (int32) (*n);
		(*n) = 0;
		entity_list.QueueClients(this, p);
		safe_delete(p);
	}
```

**Test Code**

```cpp
void Mob::TestNimbus()
{
	nimbus_effect1 = 10234;
	nimbus_effect2 = 10235;
	nimbus_effect3 = 10236;

	LogInfo("Nimbus effect before 1 [{}] 2 [{}] 3 [{}]", nimbus_effect1, nimbus_effect2, nimbus_effect2);

	std::vector<uint32*> nimbi = {&nimbus_effect1, &nimbus_effect2, &nimbus_effect3};

	for (auto &n: nimbi) {
		auto p  = new EQApplicationPacket(
			OP_RemoveNimbusEffect,
			sizeof(RemoveNimbusEffect_Struct)
		);
		auto *r = (RemoveNimbusEffect_Struct *) p->pBuffer;
		r->spawnid       = GetID();
		r->nimbus_effect = (int32) (*n);
		(*n) = 0;
		entity_list.QueueClients(this, p);
		safe_delete(p);
	}

	LogInfo("Nimbus effect after 2 [{}] 2 [{}] 3 [{}]", nimbus_effect1, nimbus_effect2, nimbus_effect2);
}
```

Result

```
[Zone] [Info] Nimbus effect before 1 [10234] 2 [10235] 3 [10235]
[Zone] [Info] Nimbus effect after 2 [0] 2 [0] 3 [0]
```